prov/gni/src/gnix_rma.c

```diff
@@ -140,13 +184,15 @@ static int __gnix_rma_post_err(struct gnix_tx_descriptor *txd)
 
 	req->tx_failures++;
 	if (req->tx_failures < req->gnix_ep->domain->params.max_retransmits) {
+		_gnix_nic_tx_free(req->gnix_ep->nic, txd);
+
```

Comment:
I started to dislike the idea of holding a TXD in a queued request.  Previously, I added the req->txd field to hold a reference to a TXD previously used by a request.  I thought it might be convenient to avoid freeing a previously used TXD when it would be reused shortly after.  That led me to adding a good amount of extra code tracking the state of a request's TXD.  At this point it seems like it's not worth the effort accounting the req->txd field and we should just be consistent in allocating a TXD when a transaction is initiated and freeing it once we receive a completion.  I was also a bit worried that caching a TXD in a request could lead to deadlock in the VC queues.  This change frees the request after a completion.  A new TXD will be allocated when the request is retransmitted.